pkg/controllers/nodetemplate/controller.go

```diff
@@ -81,13 +89,16 @@ func (c *Controller) resolveSubnets(ctx context.Context, nodeTemplate *v1alpha1.
 	if err != nil {
 		return err
 	}
+	if len(subnetList) == 0 {
```

Comment:
Don't we already log for this? https://github.com/aws/karpenter/blob/main/pkg/providers/subnet/subnet.go#L75

Even if it doesn't error for len == 0, I say we should still move it to those files rather than here, so that it errors for every call. Same in SecurityGroups too.

Reply:
If we're going to error like this, we should probably implement an error type like `SubnetsNotFound` that we can use to do error typing. The main issue here was that we shouldn't do an error requeue on this kind of an error since it's user configuration based and retry won't help